### ISSUE TITLE
Microsoft.Win32.Registry shouldn't be supported on uap

### ISSUE THREAD

**Issue by joperezr** [dotnet/corefx#15966]
Today we have Microsoft.Win32.Registry in the closure of uap because the implementation of System.Private.Xml and System.Diagnostics.Process depend on it. We should break those dependencies at least for UAP so that we can safely remove the uap Configuration for registry.

To do:
 - [X] Remove Registry dependency from System.Private.XML (PR #15973)
 - [ ] Remove Registry dependency from System.Diagnostics.Process

cc: @Priya91 (for System.Diagnostics.Process) and @sepidehMS (for System.Private.Xml and Microsoft.Win32)

**Comment by sepidehMS** [comment]
System.Private.Xml dependency removed by https://github.com/dotnet/corefx/pull/15973

**Comment by joperezr** [comment]
@Priya91 is this something you are planning to do soon? I mean breaking the Win32.Registry dependency from Process. If not, then somebody else should work on it since it needs to be done before shipping uap

**Comment by Priya91** [comment]
@joperezr Is registry not a supported feature for uwp? Process uses registry to access process information on remote machines, and to get available verbs for a file extension from registry.

**Comment by joperezr** [comment]
no, Registry is not available, so we have to decide to either throw not supported for those set of Apis or to implement them differently. I believe that in UAP it doesn't really make sense to access process information of a remote machine BTW, so probably just throwing in that case is fine. cc @weshaggard

**Comment by weshaggard** [comment]
>so probably just throwing in that case is fine

Hopefully they won't need to throw but instead become no-ops.

**Comment by Priya91** [comment]
This is to be tracked for 2.1.0 work backlog, it's not a priority for 2.0, so i'll not be working on this for now. The change is quite elaborate, we need to separate code that uses registry and setup separate build configuration for uap on process, i am not sure how to do this with the latest infra changes.

**Comment by danmosemsft** [comment]
I agree you have higher priority work. 

However it is only used in 2 places, perf counters and ProcessStartInfo.Verbs. Both could quickly be wrapped in `if !NETNATIVE` temporarily and return null or throw or empty list. Then I assume adding the new configuration is quick if you know what you are doing. @joperezr if this would be 10 mins for you then you could do that.

**Comment by danmosemsft** [comment]
Better than adding `#if !uap` woudl be to fail at runtime -- catch MissingMethodException. That's definitely OK with ILC but not sure our build process can accommodate doing that.

**Comment by ViktorHofer** [comment]
@danmosemsft I don't think that we can solve this without targeting uap specifically (with #if uap) **till 2.0**. Should we defer it to 2.1?

**Comment by weshaggard** [comment]
For this I think we will need a uap configuration. I would really like to avoid depending on MissingMethodExceptions for any real scenarios.

**Comment by danmosemsft** [comment]
@viktorhofer seems we still have
```
    <!-- 
        Temporarily necessary as this is only being built for UWP due to an unwanted dependency on it 
        https://github.com/dotnet/corefx/issues/15966
      -->
    <EnablePinvokeUWPAnalyzer>false</EnablePinvokeUWPAnalyzer>
```

**Comment by ViktorHofer** [comment]
@danmosemsft disabling the analyzer is still necessary as Microsoft.Win32.Registry has a shipping UWP compatible configuration (netstandard). Though, the comment needs to be removed.
